llms/mlx_lm/models/llama.py

```diff
@@ -154,6 +154,9 @@ def __call__(
         mask = None
         if h.shape[1] > 1:
             mask = nn.MultiHeadAttention.create_additive_causal_mask(h.shape[1])
+            if cache is not None:
+                pad_size = cache[0][0].shape[2]
+                mask = mx.pad(mask, ((0, 0), (0, pad_size)))
```

Comment:
This seems like a useful change but it would be better to unify it across models.

Maybe we can have a new `create_additive_causal_mask` in the `base.py` file: https://github.com/ml-explore/mlx-examples/blob/main/llms/mlx_lm/models/base.py and it takes two variables instead of one? Then we can use it in all the model files instead of `create_additive_causal_mask`?

Reply:
That seems pretty reasonable to me. Do you have any thoughts on how it's implemented? I noticed the speculative decoding example also includes this behavior however in a different implementation. Happy to work with any implementation if you have any preferences. https://github.com/ml-explore/mlx-examples/blob/main/llms/speculative_decoding/model.py#L256

Reply:
Actually, would this change be better suited to upstream into mlx itself? https://github.com/ml-explore/mlx/blob/ec8578d41a49d0dd4dc966a2a9e5ef9edca9f172/python/mlx/nn/layers/transformer.py#L102

Reply:
Oh nice, the one in the spec decoding example looks quite reasonable. I'm not sure if it is faster than doing the square + padding as you have it here. Probably either is fine as it's a very small part of the model.